packages/cactus-plugin-satp-hermes/src/main/typescript/core/stage-services/client/stage2-client-service.ts

```diff
@@ -323,22 +322,27 @@ export class Stage2ClientService extends SATPService {
             ledgerType: sessionData.senderAsset.networkId?.type as LedgerType,
           } as NetworkId;
 
-          const token: FungibleAsset = protoToAsset(
+          const token: Asset = protoToAsset(
             sessionData.senderAsset,
             networkId,
-          ) as FungibleAsset;
+          ) as Asset;
 
           if (token.id == undefined) {
             throw new TokenIdMissingError(fnTag);
           }
 
-          if (token.amount == undefined) {
-            throw new AmountMissingError(fnTag);
+          if (!("amount" in token) && !("uniqueDescriptor" in token)) {
```

Comment:
Please have a couple functions in `satp-utils` that does the validation of these parameters across all stages and all services. Otherwise, we will bloat the code with validation code that could be imported from a validation-specific file or set of functions